### PR TITLE
Only add OS variables if they are needed

### LINKED ISSUE #9775: Inappropriate OpenStack warning when using AWS
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**

Version 1.19.0-alpha.2 (git-552bf7af1c)
(off on a feature branch)

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

1.19.0-rc.4

**3. What cloud provider are you using?**

AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**

`.build/local/kops update cluster --yes`

**5. What happened after the commands executed?**

```
[REDACTED]
I0817 21:01:52.602219   65514 executor.go:103] Tasks: 0 done / 92 total; 52 can run
I0817 21:01:54.754626   65514 executor.go:103] Tasks: 52 done / 92 total; 16 can run
I0817 21:01:55.670427   65514 executor.go:103] Tasks: 68 done / 92 total; 21 can run
W0817 21:01:57.005055   65514 bootstrapscript.go:125] exporting username and password. Consider using application credentials instead.
W0817 21:01:57.045083   65514 bootstrapscript.go:125] exporting username and password. Consider using application credentials instead.
W0817 21:01:57.069689   65514 bootstrapscript.go:125] exporting username and password. Consider using application credentials instead.
I0817 21:01:57.876201   65514 executor.go:103] Tasks: 89 done / 92 total; 3 can run
I0817 21:02:00.532525   65514 executor.go:103] Tasks: 92 done / 92 total; 0 can run
```

**6. What did you expect to happen?**

No scary warnings about exporting username and password.

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

N/A

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**

N/A

**9. Anything else do we need to know?**

The warning comes from Openstack-specific code, looking at Openstack-specific environment variables. It should not be printed when not using the Openstack cloud provider.

/assign @olemarkus

### PR DESCRIPTION
**Pull request by olemarkus**
Fixes #9775

### CODE REVIEW

File: pkg/model/bootstrapscript.go [hunk 1]
```diff
@@ -511,3 +514,27 @@ func (b *BootstrapScript) createProxyEnv(ps *kops.EgressProxySpec) string {
 	}
 	return buffer.String()
 }
+
+func usesOpenStack(cluster *kops.Cluster) bool {
+	if kops.CloudProviderID(cluster.Spec.CloudProvider) == kops.CloudProviderOpenstack {
```
Comment (hakman) [inline pkg/model/bootstrapscript.go:519]: @olemarkus Why isn't this enough to decide that the cloud provider is OpenStack?

Comment (olemarkus) [reply pkg/model/bootstrapscript.go:519]: Technically you could run on AWS, but use swift as configbase. No one is probably doing this though.
The use case is common though, if you openstack as provider, using S3 as configbase is common.

Comment (hakman) [reply pkg/model/bootstrapscript.go:519]: You are using this function to check if the cluster is OS.
Why is it not enough to check `cluster.Spec.CloudProvider`?

**Pull request merged**